jupyter_dashboards/nbextension/notebook/dashboard-view/layout/grid/layout.js

```diff
@@ -627,9 +500,22 @@ define([
             }
             $cell.attr(attrs);
 
+            // add widget to Gridstack, reject if it takes too long
+            var rejectTimeout = window.setTimeout(function() {
+                cellAddedPromise.reject('Timeout waiting for Gridstack to add widget', $cell);
+            }, 10000);
+            var onChange = function(event, elements) {
+                if (elements.some(function(el) { return el.el.is($cell); })) {
+                    cellAddedPromise.resolve($cell);
+                    window.clearTimeout(rejectTimeout);
+                    self.gridstack.container.off('change', onChange);
+                }
+            };
+            this.gridstack.container.on('change', onChange);
```

Comment:
`off` is getting passed the callback function, so namespacing is not necessary. Additionally, this function gets called many times in certain situations, so namespacing would get complex.